### DISCUSSION TITLE
DIA-NN searching non-standard PTMs

### DISCUSSION THREAD

**Discussion by zcbtlmu** [vdemichev/DiaNN#930] [General]
Hi,

I have lots of DDA LC-MS data with non-canonical PTMs and have used this to generate a spectral library to search against a DIA LCMS run. When adding all DDA files (to generate spectral library) and my DIA file, with the specific modifications that I am interested in, and adding the variable modification in the MSFragger tab, the search crashed: 

Process 'DIA-NN' finished, exit code: -1
Process returned non-zero exit code, stopping
ERROR: C:\diann\src\diann.cpp: 2774: unknown modification: 

Is it possible to search for modifications other than standard ones with DIA-NN and DDA data containing the modification of interest?

Thanks

**Comment by vdemichev** [comment]
DIA-NN is not supposed to be used for DDA data. It works (accidentally :)) for some DDA data but not other data. So better make a lib with MSFragger and then search with this lib using DIA-NN, in separate steps.
How does the whole log look like? In general, when using a spectral library DIA-NN does not care about not knowing modifications

**Comment by pahellwi** [comment]
@vdemichev I have a similar problem. I have created a library with Fragipipe and used it in DIA-NN for DIA-PASEF measurements. My library contains peptides with a modification (http://www.unimod.org/modifications_view.php?editid1=896) that cannot be selected in DIA-NN. This gives me an error code similar to zcbtlmu. I then filtered the library and removed all peptides with the modification, after which it worked without any problems. So it seems to be due to the modification. Is there a possibility that DIA-NN can use the modification? Thank you in advance.

**Reply by vdemichev** [reply]
Yes, just make sure Library generation is set to 'IDs, RT and IM profiling' and will work with arbitrary modifications in the library.

**Reply by pahellwi** [reply]
@vdemichev It worked. Thank you a lot!
